RSSReader/Presenter/SearchPresenter.m

```diff
@@ -27,41 +32,66 @@ - (void)searchFeeds:(NSString *)url array:(NSMutableArray<SearchFeedItem *> *)ar
     [array retain];
     url = [self configureURL:url];
     if ([self validateUrl:url]) {
-        @try {
-            NSString *html = [self.networkManager downloadHTML:url];
-
+        NSString *html = [self.networkManager downloadHTML:url];
+        
+        if ([html isEqualToString:@""]) {
+            [self.delegate showException:kNoSuchSiteMessage];
+            completion();
+        } else {
             HTMLParser *htmlParser = [HTMLParser new];
-            [htmlParser parseHTML:html array:(NSMutableArray<SearchFeedItem *> *)array];
+            [htmlParser parseHTML:html array:array sitename:url];
             [htmlParser release];
-                if (array.count == 0) {
-                    [self.delegate showException:@"No RSS feeds"];
-                }
-                completion();
-        } @catch (NSException *exception) {
-            [self.delegate showException:@"No such site"];
+            if (array.count == 0) {
+                [self.delegate showException:kNoRSSFeedsMessage];
+            }
             completion();
-        } @finally {
-
         }
     } else {
-        [self.delegate showException:@"Bad format"];
+        [self.delegate showException:kBadFormatMessage];
         completion();
     }
+    [array release];
 }
 
 - (BOOL)validateUrl: (NSString *) candidate {
-    NSString *urlRegEx = @"(http|https)://((\\w)*|([0-9]*)|([-|_])*)+([\\.|/]((\\w)*|([0-9]*)|([-|_])*))+";
+    NSString *urlRegEx = @"(http|https|Http|Https)://((\\w)*|([0-9]*)|([-|_])*)+([\\.|/]((\\w)*|([0-9]*)|([-|_])*))+";
     NSPredicate *urlTest = [NSPredicate predicateWithFormat:@"SELF MATCHES %@", urlRegEx];
     return [urlTest evaluateWithObject:candidate];
 }
 
 - (NSString *)configureURL:(NSString *)url {
-    if (![url containsString:@"https://"] && ![url containsString:@"Https://"]) {
+    if (![url containsString:@"http://"] && ![url containsString:@"Http://"] && ![url containsString:@"https://"] && ![url containsString:@"Https://"] && ![url containsString:@"HTTPS://"] && ![url containsString:@"HTTP://"] && ![url containsString:@"HTTPs://"]) {
```

Comment:
It smells.
Still not cover all cases. What about hTTps:// or any other case permutation.

Reply:
fixed